app/models/grant_submission/response.rb

```diff
@@ -136,6 +127,7 @@ def validate_by_response_type
         if document.attached?
           validate_attachment_type_if_file_upload_response
           validate_attachment_size_if_file_upload_response
+          validate_attachment_not_removed_on_submit
```

Comment:
Not sure this method should be called anytime there is an attachment. Could move the logic in the `validate_attachment_not_removed_on_submit` within this method. Otherwise could add `if submission&.user_submitted_state == 'submitted'` to avoid this getting called when saving a draft.